Civi/Core/SettingsBag.php

```diff
@@ -352,7 +352,7 @@ protected function setDb($name, $value) {
     }
     $dao->find(TRUE);
 
-    if (isset($metadata['on_change'])) {
+    if (isset($metadata['on_change']) && !($value === 0 && ($dao->value === NULL || unserialize($dao->value) === 0))) {
```

Comment:
Thanks @seamuslee001. It passes the test now but the main issue seems to be appearing after this change. As [$value is supposed to be of type mixed](https://github.com/civicrm/civicrm-core/blob/master/Civi/Core/SettingsBag.php#L325) in this function, `===` will always fail for `'0'` if passed as a string.